### PR TITLE
API Versioning: Supporting Active Model Serializers namespacing

### PR DESCRIPTION
**Pull request by syntaxTerr0r**
Adding failling spec for issue #48

### CODE REVIEW

**Comment by dblock** [comment]
Do fix the build to start, squash your commits, describe how this is problematic and describe a path for an existing user in UPGRADING.

**Comment by syntaxTerr0r** [comment]
@dblock The build seems to fail for external reason of my PR: dependencies are outdated.
`Gem::InstallError: listen requires Ruby version >= 2.2.3, ~> 2.2.`

`listen` is a dependencies of `guard-rspec`.

I can do the following: 

- fix `.travis.yml` to build against ruby 2.2 (and remove 2.1.1)
- lock down `guard-rspec` to a previous version

Which sound better to you? What's the policy here? Since it's a development dependency I think it should be kept up to date, especially because Guard officially announced to not support Ruby 2.1 anymore: https://github.com/guard/guard#readme

**Comment by syntaxTerr0r** [comment]
Okay, anything legacy:

- 1.9.3
- rbx-2.2.10
- jruby-19mode

... won't pass build.

**Comment by dblock** [comment]
I took care of this in https://github.com/jrhe/grape-active_model_serializers/pull/50, you can rebase against master.

**Comment by syntaxTerr0r** [comment]
Okay, this seems to be okay now.

File: README.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ Use [active_model_serializers](https://github.com/rails-api/active_model_seriali
 [![Build Status](https://api.travis-ci.org/jrhe/grape-active_model_serializers.png)](http://travis-ci.org/jrhe/grape-active_model_serializers) [![Dependency Status](https://gemnasium.com/jrhe/grape-active_model_serializers.png)](https://gemnasium.com/jrhe/grape-active_model_serializers) [![Code Climate](https://codeclimate.com/github/jrhe/grape-active_model_serializers.png)](https://codeclimate.com/github/jrhe/grape-active_model_serializers)
 
 ## Breaking Changes
```
Comment (dblock) [inline README.md]: Lets move this into an UPGRADING document like https://github.com/ruby-grape/grape/blob/master/UPGRADING.md?

File: README.md [hunk 2]
```diff
@@ -78,6 +80,83 @@ end
 # root = people
 ```
 
+### API versioning
+
+If you haven't declared an API version in Grape, nothing change.
```
Comment (dblock) [inline README.md]: nothing _changes_

**Comment by dblock** [comment]
I'll merge this and move the CHANGELOGs and other UPGRADING things around after. Thanks!

File: lib/grape-active_model_serializers/formatter.rb [hunk 3]
```diff
@@ -15,8 +15,12 @@ def call(resource, env)
         def fetch_serializer(resource, env)
           endpoint = env['api.endpoint']
           options = build_options_from_endpoint(endpoint)
+          ams_options = {}.tap do |ns|
```
Comment (dblock) [inline lib/grape-active_model_serializers/formatter.rb]: This is almost too much Ruby IMO :)

```ruby
ams_options = {}
ams_options[:namespace] = options[:version].try(:classify) if options.try(:[], :version)
```

**Pull request merged**